javalib/src/main/scala/java/lang/String.scala

```diff
@@ -288,28 +281,34 @@ final class _String()
       false
   }
 
-  def equalsIgnoreCase(string: _String): scala.Boolean = {
-    if (string == this) {
-      true
-    } else if (string == null || count != string.count) {
+  // Ported from Scala.js commit: 37df9c2ea dated: 2025-06-30
+  @inline
+  def equalsIgnoreCase(anotherString: String): scala.Boolean = {
+    val len = length()
+    if (anotherString == null || anotherString.length() != len) {
       false
     } else {
-      var o1 = offset
-      var o2 = string.offset
-      while (o1 < offset + count) {
-        val c1 = value(o1)
-        val c2 = string.value(o2)
-        o1 += 1
-        o2 += 1
-        if (c1 != c2 && toUpperCase(c1) != toUpperCase(c2) &&
-            toLowerCase(c1) != toLowerCase(c2)) {
+      var i = 0
+      while (i != len) {
+        if (caseFold(this.charAt(i)) != caseFold(anotherString.charAt(i)))
```

Comment:
This one uses the index directly but should we use offsets and `value(i)` here as well?